datafusion/core/src/physical_optimizer/enforce_distribution.rs

```diff
@@ -1908,7 +1900,7 @@ pub(crate) mod tests {
         let distribution_context = DistributionContext::new(plan);
         let mut config = ConfigOptions::new();
         config.execution.target_partitions = target_partitions;
-        config.optimizer.enable_round_robin_repartition = false;
+        config.optimizer.enable_round_robin_repartition = true;
```

Comment:
These tests expect roundrobin partitioning is added, so this config should be enabled.